api/v1alpha1/inferencepool_types.go

```diff
@@ -59,8 +59,85 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig configures the extension that runs the endpoint picking service.
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
+}
+
+// ExtensionConfig specifies how to configure an extension that runs the endpoint picker.
+type ExtensionConfig struct {
+	// ExtensionRef is a reference to a service extension.
+	ExtensionRef *ExtensionReference `json:"extensionRef"`
+
+	// ExtensionConnection configures the connection between the gateway and the extension.
+	ExtensionConnection `json:"extensionConnection"`
```

Comment:
Why is this an embedded field?

Reply:
Since this is an embedded field, should `inline` be used in the json tag? From the [json serialization pkg](https://pkg.go.dev/github.com/go-json-experiment/json):

> inline: The "inline" option specifies that the JSON representable content of this field type is to be promoted as if they were specified in the parent struct. It is the JSON equivalent of Go struct embedding. A Go embedded field is implicitly inlined unless an explicit JSON name is specified...

This would be consistent with `TypeMeta`.

Reply:
good catch, done.